src/ErrorProne.NET.StructAnalyzers/ExplicitInParameterAnalyzer.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using System.Collections.Immutable;
+
+namespace ErrorProne.NET.StructAnalyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class ExplicitInParameterAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = DiagnosticIds.ExplicitInParameterDiagnosticId;
+
+        private static readonly string Title = "Pass arguments for 'in' parameters explicitly";
+        private static readonly string MessageFormat = "Argument for parameter '{0}' should be passed explicitly";
+        private static readonly string Description = "Pass arguments for 'in' parameters explicitly";
+        private const string Category = "Performance";
```

Comment:
nit: I don't think this is performance category, right? It more like readability aspect.

Reply:
Do you have a list of categories?

Reply:
I don't have one, but for some reason I was thinking that the list should be more or less stable across all the analyzers.

Quickly looking up at [https://index](https://index) ([http://index/?leftProject=Microsoft.CodeAnalysis&leftSymbol=imvodw0ebqzf&file=Diagnostic%5CDiagnosticDescriptor.cs](http://index/?leftProject=Microsoft.CodeAnalysis&leftSymbol=imvodw0ebqzf&file=Diagnostic%5CDiagnosticDescriptor.cs)) I think in this case "Usage" can an appropriate category.

---
In reply to: [273203896](https://github.com/SergeyTeplyakov/ErrorProne.NET/pull/127#discussion_r273203896) [](ancestors = 273203896)

Reply:
Updated to 'Usage'